WalletWasabi/Services/UpdateManager.cs

```diff
@@ -13,26 +13,21 @@
 using WalletWasabi.Microservices;
 using WalletWasabi.Models;
 using WalletWasabi.Tor.Http;
+using WalletWasabi.WebClients.Wasabi;
 
 namespace WalletWasabi.Services;
 
-public class UpdateManager : IDisposable
+public class UpdateManager
 {
-	private const byte MaxTries = 2;
 	private const string ReleaseURL = "https://api.github.com/repos/zkSNACKs/WalletWasabi/releases/latest";
 
-	public UpdateManager(string dataDir, bool downloadNewVersion, IHttpClient httpClient, UpdateChecker updateChecker)
+	public UpdateManager(string dataDir, bool downloadNewVersion, IHttpClient httpClient, WasabiClient sharedWasabiClient)
```

Comment:
Please rename the httpClient to something less generic like `DownloaderHttpClient`, `GithubHttpClient` sor something that let the devs to know what that thing is for,